src/main/pages/che-6/portable-workspaces/factories_json_reference.adoc

```diff
@@ -91,87 +88,106 @@ factory.workspace.project : {
-  "until"     : EPOCHTIME,            // Factory works only before this date
-  "create"    : [perClick | perUser]  // Create one workpace per click, user or account
+  "referer"   : STRING,                 <1>
+  "since"     : EPOCHTIME,              <2>
+  "until"     : EPOCHTIME,              <3>
+  "create"    : [perClick | perUser]    <4>
 }
 ----
+<1> Works only for clients from referer.
+<2> Factory works only after this date.
+<3> Factory works only before this date.
+<4> Create one workpace per click, user, or account.
 
 [id="factories_json_reference-limitations"]
 == Limitations
 
-You can use `since : EPOCHTIME`, `until : EPOCHTIME` and `referer` as a way to prevent the Factory from executing under certain conditions. `since` and `until` represent a valid time window that will allow the Factory to activate. The `referer` will check the hostname of the acceptor and only allow the Factory to execute if there is a match.
+You can use `since : EPOCHTIME`, `until : EPOCHTIME`, and `referer` as a way to prevent the factory from executing under certain conditions. `since` and `until` represent a valid time window that will allow the factory to activate. The `referer` will check the hostname of the acceptor and only allow the factory to execute if there is a match.
 
 [id="factories_json_reference-multiplicity"]
 == Multiplicity
 
-If `create : perClick` is used, then every click of the Factory URL will generate a new workspace, each with its own identifier, name and resources. If `create : perUser` is used, then only one workspace will be generated for each unique user that clicks on the Factory URL. If the workspace has previously been generated, we will reopen the existing workspace.
+If `create : perClick` is used, then every click of the factory URL will generate a new workspace, each with its own identifier, name, and resources. If `create : perUser` is used, only one workspace is generated for each unique user that clicks on the factory URL. If the workspace has previously been generated, the existing workspace is reopened.
 
 [id="factories_json_reference-ide-customization"]
-== IDE Customization
+== IDE customization
 
 [source,json]
 ----
-factory.ide.{event} : {          // event = onAppLoaded, onProjectsLoaded, onAppClosed
-  "actions" : [{}]               // List of IDE actions to execute when event triggered
+factory.ide.{event} : {                 <1>
+  "actions" : [{}]                      <2>
 }
 
 factory.ide.{event}.actions : [{
-  "id"         : String,         // Action for IDE to perform when event triggered
-  properties : {}                // Properties to customize action behavior
+  "id"         : String,                <3>
+  properties : {}                       <4>
 }]
 ----
+<1> event = `onAppLoaded`, `onProjectsLoaded`, `onAppClosed`.
+<2> List of IDE actions to be executed when the event is triggered.
+<3> Action for the IDE to be performed when the event is triggered.
```

Comment:
I'd recommend either "action for the IDE to perform" or "action to be performed by the IDE" here.